File: Sources/ATTNAddToCartEvent.m
//
//  ATTNAddToCartEvent.m
//  attentive-ios-sdk
//
//  Created by Wyatt Davis on 1/17/23.
//

#import "ATTNEvent.h"
#import "ATTNAddToCartEvent.h"

NS_ASSUME_NONNULL_BEGIN

@implementation ATTNAddToCartEvent

- (instancetype)initWithItems:(NSArray<ATTNItem *> *)items {
    if (self = [super init]) {
        self->_items = items;
    }
    
    return self;
}

@end

NS_ASSUME_NONNULL_END

File: Sources/Public/Events/ATTNAddToCartEvent.swift
//
//  ATTNAddToCartEvent.swift
//  attentive-ios-sdk-framework
//
//  Created by Vladimir - Work on 2024-05-30.
//

import Foundation

@objc(ATTNAddToCartEvent)
public final class ATTNAddToCartEvent: NSObject, ATTNEvent {
  @objc public let items: [ATTNItem]
  @objc public var deeplink: String?

  @objc(initWithItems:)
  public convenience init(items: [ATTNItem]) {
    self.init(items: items, deeplink: nil)
  }

  @objc(initWithItems:deeplink:)
  public init(items: [ATTNItem], deeplink: String?) {
    self.items = items
    self.deeplink = deeplink
    super.init()
  }

  private override init() {
    fatalError("init() has not been implemented")
  }
}

extension ATTNAddToCartEvent: ATTNDeeplinkHandling { }
